modules/roles/server/controllers/roles.controller.js

```diff
@@ -1,27 +1,43 @@
-var mongoose = require('mongoose');
+/* promise.done(onSuccess, onError)
+ simply allows you to process resolved value. An additional benefit is that does not imply any error swallowing (as it is the case with promise.then()), it guarantees that any involved exception would be exposed. It also effectively ends the chain and does not return any further promise.
 
-var Roles = mongoose.model('Roles');
 
+/*
+ * Roles Controller
+ * Provices andvanced role functionality and API endpoints
+ */
+
+// ---------------------------- Imports ----------------------------
 var q = require('q');
+var mongoose = require('mongoose');
+//use q promise lib for mongoose promises
+mongoose.Promise = require('q').Promise;
+var Roles = mongoose.model('Roles');
 var path = require('path');
 var config = require(path.resolve('config/config'));
 var _ = require('lodash');
 
+
 // ---------------------------- Module Definition ----------------------------
-function roleModule(logger, userModule) {
+function roleModule(logger, userModule)
+{
 
-  /*
-  * check to make sure there exists an admin role with parent set to null
-  * if one doesn't exist, create it.
-  */
-  var adminCount = Roles.count({_id: 'admin', parent: null}, (err, count) =>{
-    if(count < 1)
+  // check to make sure there exists an admin role with parent set to null
+  //if one doesn't exist, create it.
+  var adminCount = Roles.count(
+  {
+    _id: 'admin',
+    parent: null
+  }, (err, count) =>
+  {
+    if (count < 1)
     {
       var role = new Roles();
       role._id = 'admin';
       role.parent = null;
-      role.save((err, data) => {
-        if(err)
+      role.save((err, data) =>
+      {
+        if (err)
```

Comment:
This actually would be a situation for logging a critical error.  If there isn't a admin user then the application is going to have serious issues.